### PR TITLE
Add some wasm simd using benchmarks

### PR DESCRIPTION
**Pull request by alexcrichton**
I added a hex encoder (not that interesting) using simd (maybe a little interesting?) which I've been using to drive some recent Cranelift improvements. Additionally I've updated the meshoptimizer benchmark to the latest version of the upstream repository and additionally updated it to use SIMD within the binary as well.

### CODE REVIEW

File: benchmarks/hex-simd/rust-benchmark/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    {
+        if true {
+            return unsafe { hex_encode_simd128(src, dst) };
+        }
+    }
+
+    hex_encode_fallback(src, dst)
+}
+
+#[target_feature(enable = "avx2")]
```
Comment (abrown) [inline benchmarks/hex-simd/rust-benchmark/src/main.rs:44]: Do we need to attribute this to another project?

**Review by abrown — APPROVED** [review]
LGTM!

File: benchmarks/hex-simd/rust-benchmark/src/main.rs (hunk 1 above)
Comment (alexcrichton) [reply benchmarks/hex-simd/rust-benchmark/src/main.rs:44]: The original source [comes from here](https://github.com/rust-lang/stdarch/blob/master/examples/hex.rs) of which I believe I'm the main substantive author of. Should I copy in the LICENSE file here to be safe though?

Comment (abrown) [reply benchmarks/hex-simd/rust-benchmark/src/main.rs:44]: Heh, it's up to you I guess!

**Pull request merged**